### PR TITLE
Add FLAIR tutorial

### PR DESCRIPTION
**Pull request by cafelton**
Thanks for the opportunity to contribute, your data is a great boon to the long-read RNA-seq field. By the way, I noticed that the way your other tutorials and templates install the aws cli does not work for me, so I used pip install awscli instead.

### CODE REVIEW

File: docs/SG_NEx_FLAIR_tutorial_notebook.ipynb [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+      "source": [
+        "## **Running software**\n",
+        "\n",
+        "Now that we have combined our reads, we can use FLAIR align to align them. If you prefer, you can align them yourself using minimap2 to align and bedtools to convert the aligned bam file to a bed file. FLAIR align just carries what we consider to be optimal parameters for long read RNA transcript alignment."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "! source activate flair && flair align -g ../reference/hg38_chr22.fa -r combined_samples.fastq -o combined_samples.flair.aligned\n",
```
Comment (cying111) [inline docs/SG_NEx_FLAIR_tutorial_notebook.ipynb:336]:
```suggestion
        "! source activate flair && flair align -g ./reference/hg38_chr22.fa -r combined_samples.fastq -o combined_samples.flair.aligned\n",
```

File: docs/SG_NEx_FLAIR_tutorial_notebook.ipynb [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+        "Next we run FLAIR correct which takes the bed file of aligned reads and corrects alignments to the annotated splice sites. If you have short read junctions, you can pass those to FLAIR with --shortread for even more precision. You can run FLAIR without the correct step, but you may find more novel transcripts that are different by only a couple of bases at the splice site, likely due to initial misalignment of reads."
+      ],
+      "metadata": {
+        "id": "miBwDn2TuuLM"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "! source activate flair && flair correct -g ../reference/hg38_chr22.fa --gtf ../reference/hg38_chr22.gtf -q combined_samples.flair.aligned.bed -o combined_samples.flair\n",
```
Comment (cying111) [inline docs/SG_NEx_FLAIR_tutorial_notebook.ipynb:357]:
```suggestion
        "! source activate flair && flair correct -g ./reference/hg38_chr22.fa --gtf ./reference/hg38_chr22.gtf -q combined_samples.flair.aligned.bed -o combined_samples.flair\n",
```

File: docs/SG_NEx_FLAIR_tutorial_notebook.ipynb [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+        "prefix.combined.isoform.read.map.txt - all detected isoforms associated with the reads that support them\n"
+      ],
+      "metadata": {
+        "id": "RaT9ocyqvSnG"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "! source activate flair && flair collapse -g ../reference/hg38_chr22.fa --gtf ../reference/hg38_chr22.gtf -q combined_samples.flair_all_corrected.bed -r combined_samples.fastq --annotation_reliant generate --generate_map --check_splice --stringent --output combined_samples.flair.collapse\n",
```
Comment (cying111) [inline docs/SG_NEx_FLAIR_tutorial_notebook.ipynb:395]:
```suggestion
        "! source activate flair && flair collapse -g ./reference/hg38_chr22.fa --gtf ./reference/hg38_chr22.gtf -q combined_samples.flair_all_corrected.bed -r combined_samples.fastq --annotation_reliant generate --generate_map --check_splice --stringent --output combined_samples.flair.collapse\n",
```

File: docs/SG_NEx_FLAIR_tutorial_notebook.ipynb [hunk 4]
```diff
@@ -0,0 +1,592 @@
...
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "sDq22ZdifaBW"
+      },
+      "outputs": [],
+      "source": [
+        "##This actually installs FLAIR\n",
```
Comment (cying111) [inline docs/SG_NEx_FLAIR_tutorial_notebook.ipynb:140]:
```suggestion
        "##This actually installs FLAIR (note the installation takes about  7.5 mins)\n",
```

File: docs/SG_NEx_FLAIR_tutorial_notebook.ipynb [hunk 5]
```diff
@@ -0,0 +1,592 @@
...
+      "metadata": {
+        "id": "LwFiAl5Sg7gZ"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Next we run FLAIR correct which takes the bed file of aligned reads and corrects alignments to the annotated splice sites. If you have short read junctions, you can pass those to FLAIR with --shortread for even more precision. You can run FLAIR without the correct step, but you may find more novel transcripts that are different by only a couple of bases at the splice site, likely due to initial misalignment of reads."
```
Comment (cying111) [inline docs/SG_NEx_FLAIR_tutorial_notebook.ipynb:348]:
```suggestion
        "Next we run FLAIR correct which takes the bed file of aligned reads and corrects alignments to the annotated splice sites. If you have short read junctions, you can pass those to FLAIR with --shortread for even more precision. You can run FLAIR without the correct step, but you may find more novel transcripts that are different by only a couple of bases at the splice site, likely due to initial misalignment of reads.
        Note that this step takes about 7.5 mins."
```

**Review by cying111 — COMMENTED** [review]
Hi @cafelton , there is only some issue in directing to the subset files, where I first could not get it run. I have suggested my solution to those code lines and added a few note sentences to indicate the slower code chunks so that users have a running time expectation for those parts.

Other than that, everything reads very nicely to me and the tutorial is quite comprehensive and very instructive.

Thank you
Ying

**Comment by cafelton** [comment]
Thanks for looking over it! I accepted all changes and I think that your
adding timing expectations will help a lot.

Best,
Colette Felton
Brooks Lab
Biomolecular Engineering and Bioinformatics
UCSC

**Pull request merged**